backend/src/main/java/com/htecgroup/skynest/util/JwtUtils.java

```diff
@@ -0,0 +1,212 @@
+package com.htecgroup.skynest.util;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.JWTVerifier;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.exceptions.JWTVerificationException;
+import com.auth0.jwt.interfaces.Verification;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.util.Date;
+
+@Component
+@Log4j2
+public class JwtUtils {
+  @Value("${backend.app.jwtSecret}")
+  private String jwtSecret;
+
+  @Value("${backend.app.jwtExpirationMs}")
+  private int jwtExpirationMs;
+
+  @Value("${backend.app.emailConfirmationLink}")
+  private String emailConfirmationLink;
+
+  @Value("${backend.app.passwordResetLink}")
+  private String passwordResetLink;
+
+  public String generateJwtEmailToken(String email) {
+    // Check for new version of JWT
+    Date now = new Date();
+    return JWT.create()
+        .withSubject(email)
+        .withIssuedAt(now)
+        .withExpiresAt(new Date(now.getTime() + jwtExpirationMs))
+        .sign(Algorithm.HMAC256(jwtSecret));
+  }
+
+  public String getEmailFromJwtEmailToken(String token) {
+    return JWT.require(Algorithm.HMAC256(jwtSecret))
+        .build()
+        .verify(token)
+        .getSubject();
+  }
+
+  public void validateJwtToken(String token) {
+    try {
+      Verification verification = JWT.require(Algorithm.HMAC256(jwtSecret));
+      Date expiresAt = verification.build().verify(token).getExpiresAt();
+      JWTVerifier verifier = verification.acceptExpiresAt(expiresAt.getTime()).build();
+      verifier.verify(token);
+    } catch (JWTVerificationException e) {
+      log.error("Invalid JWT token: {}", e.getMessage());
```

Comment:
Throw exception here, i believe that you need to return error to end user if token is not valid ?

Reply:
Makes sence, so the front-end can show them a pop-up or something.